gensim/models/nmf.py

```diff
@@ -458,26 +494,18 @@ def _setup(self, corpus):
             )
         )
 
-        is_great_enough = self._W > self.w_std * self.sparse_coef
-
-        self._W *= is_great_enough | ~is_great_enough.all(axis=0)
-
-        self._W = scipy.sparse.csc_matrix(self._W)
-
-        self.A = scipy.sparse.csr_matrix((self.num_topics, self.num_topics))
-        self.B = scipy.sparse.csc_matrix((self.num_tokens, self.num_topics))
+        self.A = np.zeros((self.num_topics, self.num_topics))
+        self.B = np.zeros((self.num_tokens, self.num_topics))
 
     def update(self, corpus, chunks_as_numpy=False):
         """Train the model with new documents.
 
         Parameters
         ----------
-        corpus : iterable of list(int, float)
-            Training corpus.
+        corpus : iterable of list of (int, float), optional
+            Training corpus. Contains list of counts of words for every document.
         chunks_as_numpy : bool, optional
-            Whether each chunk passed to the inference step should be a numpy.ndarray or not. Numpy can in some settings
-            turn the term IDs into floats, these will be converted back into integers in inference, which incurs a
-            performance hit. For distributed computing it may be desirable to keep the chunks as `numpy.ndarray`.
+            Deprecated.
```

Comment:
not better to delete it all?  yeah, it's not backward-compatible, but it's a fresh module.

Reply:
I agree. @menshikh-iv, what do you think? I bet there are not so many people who are already actively using current NMF in their projects

Reply:
I think it would be better to break compatibility now, than drag it till the 4.0.0

Reply:
not sure, let's ask @piskvorky: we want to remove some parameters from constructor, is it OK to break backward compatibility now (instead of stay it as "dead" parameters forever)?

Reply:
Current NMF wasn't promoted yet, so backward compatibility is no issue. I'd prefer the release (after docs & bug fixes) to be clean & clear.